packages/sync/src/modules/Terms.php

```diff
@@ -35,29 +35,34 @@ public function name() {
 	}
 
 	/**
-	 * Allows WordPress.com servers to retrieve a term object via the sync API.
+	 * Allows WordPress.com servers to retrieve term-related objects via the sync API.
 	 *
 	 * @param string $object_type The type of object.
 	 * @param int    $id          The id of the object.
 	 *
-	 * @return bool|\WP_Term
+	 * @return bool|object A WP_Term object, or a row from term_taxonomy table based on
 	 */
 	public function get_object_by_id( $object_type, $id ) {
+		global $wpdb;
+		$object = false;
 		if ( 'term' === $object_type ) {
-			$term = get_term( intval( $id ) );
+			$object = get_term( intval( $id ) );
```

Comment:
Something that is still weird here is that we still use `get_term()` for `term` objects, and that includes parts that pull from the term taxonomy table; while the `term_taxonomy` object below is a raw db row pull of a particular row. Should this one also work with raw term rows too?

Reply:
great call!

Reply:
On second though, on the server side, we will only have access to term_taxonomy_id when we calculate the checksum for term_taxonomy.